tracer/src/Datadog.Trace.Manual/Configuration/IntegrationSettings.cs

```diff
@@ -0,0 +1,94 @@
+﻿// <copyright file="IntegrationSettings.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.SourceGenerators;
+
+namespace Datadog.Trace.Configuration;
+
+/// <summary>
+/// Contains integration-specific settings.
+/// </summary>
+public sealed class IntegrationSettings
+{
+    private readonly bool? _enabledInitial;
+    private readonly bool? _analyticsEnabledInitial;
+    private readonly double _analyticsSampleRateInitial;
+    private OverrideValue<bool?> _enabledOverride = new();
+    private OverrideValue<bool?> _analyticsEnabledOverride = new();
+    private OverrideValue<double> _analyticsSampleRateOverride = new();
+
+    internal IntegrationSettings(string integrationName, bool? enabled, bool? analyticsEnabled, double analyticsSampleRate)
+    {
+        IntegrationName = integrationName;
+        _enabledInitial = enabled;
+        _analyticsEnabledInitial = analyticsEnabled;
+        _analyticsSampleRateInitial = analyticsSampleRate;
+    }
+
+    /// <summary>
+    /// Gets the name of the integration. Used to retrieve integration-specific settings.
+    /// </summary>
+    [Instrumented]
+    public string IntegrationName { get; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether
+    /// this integration is enabled.
+    /// </summary>
+    public bool? Enabled
+    {
+        [Instrumented]
+        get => _enabledOverride.IsOverridden ? _enabledOverride.Value : _enabledInitial;
```

Comment:
NIT: Could this be handled directly in the `OverrideValue` class?

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/5235/commits/fa0e4ffc3bf47c6056124eca6c523c9d8acb90f4